admin/scripts/modules/aktivity/aktivity.php

```diff
@@ -113,12 +113,21 @@
     back();
 }
 
-if (post("korekce")) {
+if (($opacnyStavKorekce = post("korekce") !== null)) {
+    $idAktivity = post('aktivitaId');
+    if (!$idAktivity) {
+        chyba('Chybí ID aktivity pro korekci.');
+        back();
+    }
     $a = Aktivita::zId(
-        id: post("aktivitaId"),
+        id: $idAktivity,
         systemoveNastaveni: $systemoveNastaveni,
     );
-    $a?->nastavKorekci(!$a->probehlaKorekce());
```

Comment:
...tak to prohodilo současný stav korekce na aktivitě - ale to nemuselo být to, co uživatel chtěl - když stránka zalaguje, tak uživatel zuřivě kliká na odeslat (nebo prostě mařkiá F5 a potvrzuje opětovné odeslání formuláře) a na backendu se korekce přepíná sem a tam, zatímco uživatel chce _"ten stav co jednou nastavil a snaží se ho uložit"_